restli-common/src/main/java/com/linkedin/restli/common/validation/RestLiDataValidator.java

```diff
@@ -608,13 +608,21 @@ private ValidationResult validateInputEntity(RecordTemplate entity)
     //  the client cannot supply them in a create request, so they should be treated as optional.
     //  similarly for update requests used as upsert (update to create), they are treated as optional.
     validationOptions.setTreatOptional(_readOnlyOptionalPredicate);
-    return ValidateDataAgainstSchema.validate(entity, validationOptions, new DataValidator(entity.schema()));
+    return ValidateDataAgainstSchema.validate(entity, validationOptions, getValidatorForInput(entity.schema()));
   }
 
   private ValidationResult validateOutputEntity(RecordTemplate entity, DataSchema validatingSchema)
   {
-    DataSchemaAnnotationValidator validator = new DataSchemaAnnotationValidator(validatingSchema);
-    return ValidateDataAgainstSchema.validate(entity.data(), validatingSchema, new ValidationOptions(), validator);
+    return ValidateDataAgainstSchema.validate(entity.data(), validatingSchema, new ValidationOptions(),
+        getValidatorForOutput(validatingSchema));
+  }
+
+  protected Validator getValidatorForOutput(DataSchema validatingSchema) {
+    return new DataSchemaAnnotationValidator(validatingSchema);
```

Comment:
Sorry can you remind me why this changes from instantiating "DataValidator" to "DataSchemaAnnotationValidator" now?

Reply:
Both before and after are instantiating a `DataSchemaAnnotationValidator`, it was just abstracted into its own method to allow for override access

Reply:
both before and after instantiate `DataSchemaAnnotationValidator`, this just abstracts it out into its own method

Reply:
you are right. I previously misunderstood